runtime/doc/lua.txt

```diff
@@ -4433,4 +4433,53 @@ tohtml.tohtml({winid}, {opt})                         *tohtml.tohtml.tohtml()*
         (`string[]`)
 
 
+==============================================================================
+Lua module: vim.net                                                  *vim.net*
+
+vim.net.download({url}, {opts})                           *vim.net.download()*
+    Asynchronously download a file
+
+    Parameters: ~
+      • {url}   (`string`) Request URL
+      • {opts}  (`table?`) Additional options
+
+                Example: >lua
+                    -- Download a file
+                    -- The file will be saved in the `cwd` with the name `anything`
+                    vim.net.download("https://httpbingo.org/anything")
+
+                    -- Download a file to a path
+                    -- The file will be saved in `/tmp/somefile`
+                    vim.net.download("https://httpbingo.org/anything", {
+                      as = "tmp/somefile",
+                    })
+
+                    -- Download a file while sending headers
+                    vim.net.download("https://httpbingo.org/anything", {
+                      headers = {
+                        Authorization = { "Bearer foo" },
+                      },
+                    })
+
+                    -- Download a file while handling basic auth
+                    vim.net.download("https://httpbingo.org/basic-auth/user/password", {
+                      credentials = "user:password",
+                    })
+<
+                • {as}? (`string`) Path to write the downloaded file to. If
+                  not provided, the one inferred from the URL will be used.
+                  Defaults to `nil`
+                • {credentials}? (`string`) Credentials with the format
```

Comment:
If these options map to `curl` options, in most cases it's probably a very good idea to use the same names as curl. That kind of entropy-reduction helps greatly for discovery and intuition. In some cases it may make sense to violate this, but otherwise it's a very good default guideline.

For example, curl names this `--user`.

```suggestion
                • {user}? (`string`) Credentials with the format
```